internal/pkg/api/handleAck.go

```diff
@@ -385,8 +377,28 @@ func (ack *AckT) handlePolicyChange(ctx context.Context, zlog zerolog.Logger, ag
 	return errors.Wrap(err, "handlePolicyChange update")
 }
 
+func (ack *AckT) invalidateAPIKeys(ctx context.Context, agent *model.Agent) {
+	var toRetire []model.ToRetireAPIKeyIdsItems
+
+	for _, out := range agent.Outputs {
```

Comment:
more of a question: is Outputs new outputs or old outputs?
what happens in case you change output in policy to different ES? will this contain old output so we can invalidate apikeys?